### ISSUE TITLE
Gracefully handle scenario when user make a commit to gitbutler/integration

### ISSUE THREAD

**Issue by Caleb-T-Owens** [gitbutlerapp/gitbutler#3972]
Ref: https://discord.com/channels/1060193121130000425/1073202153163857920/1247103722270494732

**Comment by To1ne** [comment]
I'm in a very similar situation. I use Emacs, and I just love to create commits from Emacs. So what I would like is GitButler to pick up that commit, place it in the selected "default branch" (at least that's how I understand a "default branch" should be used).

I think it would be possible by installing a post-commit hook.

**Comment by Caleb-T-Owens** [comment]
@To1ne That is a really interesting idea; I'll bring it up with the team 👍

**Comment by krlvi** [comment]
Closing this as a duplicate of #9656, which tracks the same case: the managed workspace ref being advanced by a normal Git commit, plus the remaining recovery UX.

Accidental commits are now blocked by the managed pre-commit hook added in #11973 and shipped in 0.18.8. #9656 is the better place to track recovery when the ref still gets advanced.
